### PR TITLE
[MRG] DOC added new dependencies for 2.0

### LINKED ISSUE #6580: Ensure install requirements in documentation are up to date before release
I think that at least the following may be missing
- subprocess32  #6576 
- six #6556
- functools32 #5295

### PR DESCRIPTION
**Pull request by NelleV**
Added the missing dependencies from our installation instructions.

closes #6580

### CODE REVIEW

**Comment by NelleV** [comment]
I've added a new sections for python 2 only requirements. I've also removed six as a library shipped with matplotlib from the documentation, as this is obviously not the case anymore.

File: INSTALL [hunk 1]
```diff
@@ -265,6 +268,9 @@ Optional dependencies
     required, but can make installation go more smoothly if the libraries and
     headers are not in the expected locations.
 
+`subprocess32 <https://pypi.python.org/pypi/subprocess32/>`_
```
Comment (tacaswell) [inline INSTALL]: Sigh, I even merged the PR where that went in....

**Comment by tacaswell** [comment]
Thanks for cleaning the the remnants of when we vendored `six`.  Just waiting for the docs to build on travis.

File: INSTALL [hunk 2]
```diff
@@ -214,10 +214,20 @@ Required Dependencies
...
 
-`functools32`
+Required dependencies for python 2
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+`functools32 <https://pypi.python.org/pypi/functools32>`_
     Required for compatibility if running on versions of Python before
     Python 3.2.
 
+`six <https://pypi.python.org/pypi/six>`_
```
Comment (tacaswell) [inline INSTALL]: `six` is required for both (because we import it all over the place).

**Comment by NelleV** [comment]
I've moved back six where it belongs.

**Comment by tacaswell** [comment]
I tweaked the wording a bit (as subprocess32 is not actually required).

**Comment by NelleV** [comment]
Thanks! (It looks good to me and I think you should totally merge this - no conflict of interest at all :) )

**Comment by tacaswell** [comment]
Probably should wait for travis to catch up (rst is harsh and capricious).

**Comment by NelleV** [comment]
Is it normal that only the travis-ci tests show up on that PR?

**Comment by efiring** [comment]
For some time now, Appveyor has not been running on 2.x, only on master.

**Comment by NelleV** [comment]
Then I guess this PR is ready to be merged :)

**Comment by efiring** [comment]
I added a commit using the web editor interface; I've never tried that before.  It doesn't seem to have triggered a re-test.

**Pull request merged**